scrapy/pipelines/files.py

```diff
@@ -308,7 +303,8 @@ def __init__(self, store_uri, download_func=None, settings=None):
 
     @classmethod
     def from_settings(cls, settings):
-        s3store = cls.STORE_SCHEMES['s3']
+        cls._load_storages(settings)
+        s3store = cls.STORAGES['s3']
```

Comment:
I thought about refactoring this piece as well.
Currently we get class and change it's attributes.
I think that settings the proper attributes would a responsibility of the given `S3`, `GCS`, `FS` class.
So, we could rewrite to instantiate the class giving the URI and the settings.

Reply:
+1, the current code doesn't look good, I also noticed it recently (https://github.com/scrapy/scrapy/pull/3625#issuecomment-463333661). It'd be good this in a separate PR, if possible.

Reply:
thanks @kmike, I can work on that after this PR to avoid merge conflicts.